core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -199,7 +212,9 @@ void invoke0(Channel channel, ClientRequestContext ctx,
             if (sessionProtocol == null) {
                 needsRelease = false;
                 try {
-                    res.close(ClosedSessionException.get());
+                    final UnprocessedRequestException cause = UnprocessedRequestException.get();
+                    handleEarlyException(ctx, req, cause);
```

Comment:
Seems like this line is not covered according to codecov.io report. Could you try to add a test case that covers this?